src/main/java/com/example/petbuddybackend/dto/rating/RatingDTO.java

```diff
@@ -0,0 +1,12 @@
+package com.example.petbuddybackend.dto.rating;
+
+import jakarta.validation.constraints.NotNull;
+import org.hibernate.validator.constraints.Range;
+
+public record RatingDTO(
```

Comment:
Dodałbym do RatingDTO pole caretakerId oraz customerId. Wtedy nie musimy podawać więcej parametrów przy endpoincie i zwracając RatingDTO dla frontu mamy więcej informacji o opinii.

Jeśli front nie będzie w stanie podać customerId przy zapytaniu i pewnie tak będzie wygodniej to można zostawić w endpoincie Principal, natomiast potem przy zwróceniu RatingDTO już zwrócimy razem z custmerId.

`        Optional<Integer> customerId,
        @NotNull
        Integer caretakerId,
`

Reply:
Moim zdaniem to podejście byłoby mało czytelne. Teraz mamy endpoint POST /api/caretaker/{caretakerEmail}/rating i tutaj wiesz że chodzi o dodanie opinii do zasobu caretakerEmail. Jakby to było w body to nie bardzo wiadomo by było co z tym zrobić. Plus trzeba by to było walidować w zależności od endpointu bo w jednym to jest wymagane a w drugim nie. 

Jeśli chodzi o te dodatkowe informacje do opinii, to zrobiłem RatingResponse z tymi mailami dotyczącymi klienta i opiekuna. Z kolei w requeście zostawiłbym te pola co były i nie dodawał opcjonalnego caretakerId

Reply:
git